src/native/libs/System.Native/pal_interfaceaddresses.c

```diff
@@ -100,12 +101,66 @@ static inline uint8_t mask2prefix(uint8_t* mask, int length)
     return len;
 }
 
+#if !HAVE_IFADDRS && TARGET_ANDROID
+// This structure is exactly the same as struct ifaddrs defined in ifaddrs.h but since the header
+// might not be available (e.g., in bionics used in Android before API 24) we need to mirror it here
+// so that we can dynamically load the getifaddrs function and use it.
+struct ifaddrs
+{
+	struct ifaddrs *ifa_next;
+	char *ifa_name;
+	unsigned int ifa_flags;
+	struct sockaddr *ifa_addr;
+	struct sockaddr *ifa_netmask;
+	union
+	{
+		struct sockaddr *ifu_broadaddr;
+		struct sockaddr *ifu_dstaddr;
+	} ifa_ifu;
+	void *ifa_data;
+};
+#endif
+
+#if !HAVE_GETIFADDRS && TARGET_ANDROID
+// Try to load the getifaddrs and freeifaddrs functions manually.
+// This workaround is necessary on Android prior to API 24 and it can be removed once
+// we drop support for earlier Android versions.
+static int (*getifaddrs)(struct ifaddrs**) = NULL;
+static void (*freeifaddrs)(struct ifaddrs*) = NULL;
+
+static void try_loading_getifaddrs()
+{
+    void *libc = dlopen("libc.so", RTLD_NOW);
+    if (libc)
+    {
+        getifaddrs = (int (*)(struct ifaddrs**)) dlsym(libc, "getifaddrs");
+        freeifaddrs = (void (*)(struct ifaddrs*)) dlsym(libc, "freeifaddrs");
+    }
+}
+
+static bool ensure_getifaddrs_is_loaded()
+{
```

Comment:
We could use [`android_get_device_api_level()`](https://developer.android.com/ndk/reference/group/apilevels#group__apilevels_1ga79b134cadc2dfed84268e67fd9e2dfd6) from `android/api-level.h` to make sure this is only used on Android < 24:

Reply:
Do you think that's really necessary? I think that checking `HAVE_GETIFADDRS` should be enough to make sure it's used only on API < 24.